mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h

```diff
@@ -5135,17 +5153,21 @@ struct GridwiseGemmRewritePattern : public OpRewritePattern<miopen::GridwiseGemm
 
     int64_t GemmABlockCopyThreadSliceLengths_GemmK;
     int64_t GemmABlockCopyThreadSliceLengths_GemmM;
-    assert(matrix_a_source_vector_read_dim != GemmG);
-    if (matrix_a_source_vector_read_dim == GemmK) {
+    switch (matrix_a_source_vector_read_dim) {
+    case GemmK:
       GemmABlockCopyThreadSliceLengths_GemmK = matrix_a_source_data_per_read;
       GemmABlockCopyThreadSliceLengths_GemmM =
           GemmABlockCopyNumberDataPerThread /
           GemmABlockCopyThreadSliceLengths_GemmK;
-    } else if (matrix_a_source_vector_read_dim == GemmMorN) {
+      break;
+    case GemmMorN:
       GemmABlockCopyThreadSliceLengths_GemmM = matrix_a_source_data_per_read;
       GemmABlockCopyThreadSliceLengths_GemmK =
           GemmABlockCopyNumberDataPerThread /
           GemmABlockCopyThreadSliceLengths_GemmM;
+      break;
+    case GemmG:
+      llvm_unreachable("Group matrix unimplemented");
```

Comment:
The message should be something like : " vector loads / stores are not possible on GemmG dimension ".